docs/serialization.md

```diff
@@ -0,0 +1,91 @@
+# Serialization
+
+The linker cannot analyze the patterns typically used by reflection-based serializers. Such serializers should be annotated with `RequiresUnreferencedCodeAttribute`, and using them in a trimmed app will likely not work (or will work unpredictably). Set `SuppressTrimAnalysisWarnings` to `false` to see static analysis warnings for these patterns.
```

Comment:
Is the `SuppressTrimAnalysisWarnings` part necessary now?

Reply:
I initially wanted to mention it in case people run into issues for xamarin-android apps, where warnings are suppressed by default, but I changed it to just point to the docs for warnings which should stay up-to-date wrt defaults.